packages/core/src/services/commits/runDocumentAtCommit.ts

```diff
@@ -183,11 +171,24 @@ async function doSomeCommonOperations({
   const { completed, conversation } = await chain.step(previousResponse?.text)
   const config = validateConfig(conversation.config)
   if (!apiKey || apiKey?.name !== conversation.config.apikey) {
-    apiKey = await findApiKey({ scope, name: config.apiKey })
+    apiKey = await findApiKey({ scope, name: config.provider })
   }
 
   const msgs = conversation.messages.slice(sentCount)
   sentCount += msgs.length
 
   return { sentCount, apiKey, conversation, completed, config }
 }
+
+async function findApiKey({
+  scope,
+  name,
+}: {
+  scope: ProviderApiKeysRepository
+  name: string
+}) {
+  const apiKeyResult = await scope.findByName(name)
+  if (apiKeyResult.error) throw apiKeyResult.error
```

Comment:
Just return the result no the `.error`. Also why `value!`? It can be `null`?

Reply:
nah this is for internal use no real need to return result, i want this to throw unless it works